server/.env.example

```diff
@@ -0,0 +1,4 @@
+BUILD_DIR=build
+NEPTUNE_SERVER_VERSION=v0.1.0
+BUILD_ERR_FILE=build-error.log
+BUILD_TARGET=aarch64-apple-darwin
```

Comment:
i'd love to minimize the dependency on an env file where possible. 
- `BUILD_DIR` and `BUILD_ERR_FILE` could just be defined in the makefile directly
- we should have a separate makefile command for each build target so that we don't have to modify the env file each time we want to compile to a different target. Perhaps a `make m1`, `make intel`, `make linux`, and then a `make all` that encapsulates all of them (actual names should be different)